Fixes #12106 - Show popovers correctly on interface modal

tbrisker opened the pull request:
When saving an interface the popovers were not destroyed, leading to
them breaking when opening the same interface form again.
Also, the help text for the various checkboxes has been hidden with
popover to reduce screen clutter.

dLobatog commented:
[test]

dLobatog commented:
@tbrisker Interesting! It fixes the issue well. The hidden help text admittedly looks better IMO, do you agree @ares @tstrachota ?

If you can, do you mind to write an integration test that checks the popovers work well after saving an interface and reopening it? It's kind of an obscure change that could very easily break.

Since the other part of this PR (the help text) is no-risk and already makes it better, would you submit a separate PR & issue number, etc.. with that? I think it'd be merged now already.

Thanks again!

tbrisker committed:
Fixes #12106 - Show popovers correctly on interface modal

When saving an interface the popovers were not destroyed, leading to
them breaking when opening the same interface form again.

tbrisker commented:
@dLobatog I have split the help text part to PR #2835 . However I'm not sure how to write an integration test for this - the popover is displayed but its position is miscalculated so it appears outside the screen.

shlomizadok commented:
Tested. Works very well :+1:

dLobatog commented:
@tbrisker Fine, but I'll nag you if this comes as a regression later :smile:

dLobatog commented:
Merged as 988cee422836de072606978e3454dcf53ed593dd, thanks @tbrisker & @shlomizadok !
